packages/@aws-cdk/aws-ec2/lib/user-data.ts

```diff
@@ -68,14 +121,62 @@ class LinuxUserData extends UserData {
     this.lines.push(...commands);
   }
 
+  public addOnExitCommands(...commands: string[]) {
+    this.onExitLines.push(...commands);
+  }
+
   public render(): string {
     const shebang = this.props.shebang !== undefined ? this.props.shebang : '#!/bin/bash';
-    return [shebang, ...this.lines].join('\n');
+    return [shebang, ...(this.renderOnExitLines()), ...this.lines].join('\n');
+  }
+
+  public addDownloadAndExecuteS3FileCommand( params: S3DownloadAndExecuteOptions ): void {
+    if (!this.functionsAdded.has('download_and_execute_s3_file')) {
+      this.addCommands("download_and_execute_s3_file () {\n" +
+          "local s3Path=$1;\n" +
+          "local path=$2;\n" +
+          "shift;shift;\n" +
+          "echo \"Downloading file ${s3Path} to ${path}\";\n" +
+          "mkdir -p $(dirname ${path}) ;\n" +
+          "aws s3 cp ${s3Path} ${path};\n" +
+          "if [ $? -ne 0 ]; then exit 1;fi;\n" +
```

Comment:
The script should be running under `set -e`, so this test is not necessary. (Verify that, make it true if it isn't)

Reply:
By default the scripts are not being run under `set -e` as such the new execute file command is running set -e